internal/fs/handle/file_test.go

```diff
@@ -946,7 +946,13 @@ func (t *fileTest) Test_ReadWithReadManager_FullReadSuccessWithBufferedRead() {
 
 	assert.NoError(t.T(), err)
 	assert.Equal(t.T(), fileSize, resp.Size)
-	assert.Equal(t.T(), expectedData, buf)
+	// When buffered read is enabled, the data is returned in resp.Data and not
+	// directly in the buffer. We need to copy it to compare.
+	bytesCopied := 0
+	for _, dataSlice := range resp.Data {
```

Comment:
Please create a test utility to convert from resp.Data -> string and use that everywhere. Keep the utility common place so that you can use in the Buffered-reader test too.